cmd/cluster-agent/api/agent/agent.go

```diff
@@ -149,16 +158,34 @@ func getCheckLatestEvents(w http.ResponseWriter, r *http.Request) {
 }
 
 func getPodMetadata(w http.ResponseWriter, r *http.Request) {
+	if err := apiutil.Validate(w, r); err != nil {
+		return
+	}
 	vars := mux.Vars(r)
+	var err error
+	var slcB []byte
 	nodeName := vars["nodeName"]
 	podName := vars["podName"]
-	svcList := as.GetPodServiceNames(nodeName, podName)
 
-	slcB, err := json.Marshal(svcList)
+	if podName == "*" {
+		log.Info("Computing service map on all nodes ...")
```

Comment:
You should remove "..." as it'll look like the logs bellow would be tied to this one and as we have a lot of async stuff it could be an unrelated log ;)